### PR TITLE
feat(#8311): reveal password on demand

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

Adds an icon-button to show/hide the password on the login page.

medic/cht-core#8311

**Tested scenarios:** [video](https://github.com/medic/cht-core/assets/66472237/cb34216b-fb8c-43af-9443-5ec745ee142d)

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8311_show_password_eye_icon/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8311_show_password_eye_icon/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8311_show_password_eye_icon/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/src/public/login/script.js [hunk 1]
```diff
@@ -234,17 +239,25 @@ const checkUnsupportedBrowser = () => {
...
-    document.getElementById('unsupported-browser').classList.remove('hidden');
+    document.getElementById('unsupported-browser').classList.remove(HIDDEN_CLASS);
   }
 };
 
+const togglePassword = () => {
+  const input = document.getElementById(PASSWORD_INPUT_ID);
+  input.type = input.type === 'password' ? 'text' : 'password';
+  document.getElementById('show-password').classList.toggle(HIDDEN_CLASS);
+  document.getElementById('hide-password').classList.toggle(HIDDEN_CLASS);
```
Comment (garethbowen) [inline api/src/public/login/script.js:250]: I think the code comes out slightly cleaner if you toggle a class on a parent, for example, adding a class to the `#password-container` element. If you name the class something sensible like "mask" (or a better name) then you can change the style of both the icon elements with one line of JS.

Comment (latin-panda) [reply api/src/public/login/script.js:250]: Okay. I was thinking that, or keeping the CSS minimal and reusing the classes we already had

**Comment by latin-panda** [comment]
@garethbowen @ngaruko please have a look, thanks

File: api/src/public/login/style.css [hunk 2]
```diff
@@ -147,6 +147,38 @@ form {
...
 
+#password-container {
+  position: relative;
+  height: 40px;
+}
+
+#password-container #password {
+  width: 245px;
+  padding: 6px 43px 6px 10px;
+}
```
Comment (garethbowen) [inline api/src/public/login/style.css:158]: Why is this necessary? If I delete it the password field looks the same to me...

File: api/src/public/login/style.css [hunk 3]
```diff
@@ -147,6 +147,38 @@ form {
...
+
+#password-container #password-toggle {
+  position: absolute;
+  right: 0px;
+  top: 0px;
```
Comment (garethbowen) [inline api/src/public/login/style.css:163]: Nit: Slightly cleaner to not use units for 0 values.

```suggestion
  right: 0;
  top: 0;
```

File: api/src/public/login/style.css [hunk 4]
```diff
@@ -147,6 +147,38 @@ form {
...
+  height: 100%;
+  padding: 0 10px;
```
Comment (garethbowen) [inline api/src/public/login/style.css:165]: This looks more centered to me (on Firefox and Chrome).

```suggestion
  padding: 2px 8px;
```

File: api/src/public/login/style.css [hunk 5]
```diff
@@ -147,6 +147,38 @@ form {
...
+  height: 100%;
+  display: inline-block;
+  vertical-align: middle;
+}
+
+#password-container .show-password-icon,
+#password-container.hidden-password .hide-password-icon {
+  display: none;
+}
+
```
Comment (garethbowen) [inline api/src/public/login/style.css:181]: I think lines 169 and 170 can be simplified to being a single css rule instead of two. This is purely for readability...

```suggestion
#password-container img {
  width: 24px;
  height: 100%;
  vertical-align: middle;
}

#password-container.hidden-password .show-password-icon,
#password-container .hide-password-icon {
  display: inline-block;
}

#password-container .show-password-icon,
#password-container.hidden-password .hide-password-icon {
  display: none;
}

```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice quick turnaround!

Some CSS suggestions inline...

File: api/src/public/login/style.css (hunk 2 above)
Comment (latin-panda) [reply api/src/public/login/style.css:158]: So letters don't go behind the icons
Screenshot 2023-06-15 at 9 06 16 am

File: api/src/public/login/style.css (hunk 5 above)
Comment (latin-panda) [reply api/src/public/login/style.css:181]: Done

File: api/src/public/login/style.css (hunk 2 above)
Comment (garethbowen) [reply api/src/public/login/style.css:158]: Ahhh good thinking!

**Review by garethbowen — APPROVED** [review]
Nice one!

**Comment by latin-panda** [comment]
@ngaruko, just a reminder to have a look at this. Is this coverage good?
